webapp/tests/forms/flows/test_unlock_code_revocation_flow.py

```diff
@@ -230,6 +231,26 @@ def test_if_user_exists_but_elster_returns_no_antrag_found_then_user_is_deleted(
                     self.input_step, self.render_info_input_step, self.session_data)
                 self.assertRaises(UserNotExistingError, find_user, existing_idnr)
 
+    def test_if_user_exists_but_elster_returns_already_revoked_then_user_is_deleted(self):
+        existing_idnr = '04452397687'
+        date_of_birth = '01.01.1985'
+        create_user(existing_idnr, date_of_birth, '0000')
+
+        with open('tests/samples/sample_vast_revocation_response_failure.xml') as failue_sample:
+            failure_server_response = failue_sample.read()
+        with app.app_context() and app.test_request_context(method='POST',
+                                                            data={'idnr': existing_idnr, 'dob': date_of_birth}):
+            with patch("app.forms.flows.unlock_code_revocation_flow.elster_client.send_unlock_code_revocation_with_elster") \
+                    as fun_unlock_code_revocation:
+                expected_error = ElsterRequestAlreadyRevoked()
+                expected_error.eric_response = b''
+                expected_error.server_response = failure_server_response
+                fun_unlock_code_revocation.side_effect = expected_error
+
+                render_info, _ = self.flow._handle_specifics_for_step(
+                    self.input_step, self.render_info_input_step, self.session_data)
+                self.assertRaises(UserNotExistingError, find_user, existing_idnr)
```

Comment:
Just FYI, because I had to have a second look: 

At first, I was wondering why we check for that error here. Then I realized that this is the way you want to check whether the user is still in the database. I think that this is okay in that way here, because this is also how we check it in the application.